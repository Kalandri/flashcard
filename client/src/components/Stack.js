import React from 'react';
import { connect } from 'react-redux';
import Card from './Card';

export const Stack = ({ stack: { title, cards } }) => {
  return (
    <div>
      <h3>{title}</h3>
      <br />
      {
        cards.map(card => {
          return (
            <Card key={card.id} card={card} />
          )
        })
      }
    </div>
  )
}

function mapStateToProps(state) {
  return { stack: state.stack };
}

export default connect(mapStateToProps, null)(Stack);