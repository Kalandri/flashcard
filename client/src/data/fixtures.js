export const stack = {
  id: 0,
  title: 'test title 0',
  cards: [
    {id: 0, prompt:'test01', answer: 'answer01'},
	{id: 1, prompt:'test02', answer: 'answer02'}
  ]
};

export const stack1 = {
  id: 1,
  title: 'test title 1',
  cards: [
    {id: 0, prompt:'test11', answer: 'answer11'},
	{id: 1, prompt:'test12', answer: 'answer12'}
  ]
};

export const stacks = [stack, stack1];
