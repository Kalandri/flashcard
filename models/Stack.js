const mongoose = require('mongoose');
const { Schema } = mongoose;
const CardSchema = require('./Card');

const stackSchema = new Schema({
	title: String,
	cards: [CardSchema],
	_user: { type: Schema.Types.ObjectId, ref: 'User'},
	generated: Date
});

mongoose.model('stack', stackSchema);