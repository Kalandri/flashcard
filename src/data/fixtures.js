export const stack = {
  id: 0,
  title: 'test title',
  cards: [
    {id: 0, prompt:'test1', answer: 'answer1'},
	{id: 1, prompt:'test2', answer: 'answer2'}
  ]
};
