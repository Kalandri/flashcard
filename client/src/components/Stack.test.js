import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import { Stack } from './Stack';
import { stack } from '../data/fixtures';
import renderer from 'react-test-renderer';
import '../setupTests.js';

const props = { stack };

describe('Stack', ()=> {
	const stack =shallow(<Stack {...props} />);

	it('renders the title', ()=>{
		expect(stack.find('h3').text()).toEqual(props.stack.title);
	});

	it('renders the correct number of cards', ()=>{
		expect(stack.find('Card').length).toEqual(props.stack.cards.length);
	});
});

test('snapshot test for Stack', ()=>{
	const component = renderer.create(
		<MemoryRouter >
			<Stack {...props} />
		</MemoryRouter >
	);
	const tree=component.toJSON();
	expect(tree).toMatchSnapshot();
});