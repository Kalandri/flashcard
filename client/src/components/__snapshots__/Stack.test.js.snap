// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Stack 1`] = `
<div>
  <h3>
    test title 0
  </h3>
  <br />
  <div
    className="card"
    onClick={[Function]}
  >
    <div
      className="card-prompt"
    >
      <h4>
        test01
      </h4>
    </div>
    <div
      className="card-answer"
    >
      <h4
        className="text-hidden"
      >
        answer01
      </h4>
    </div>
  </div>
  <div
    className="card"
    onClick={[Function]}
  >
    <div
      className="card-prompt"
    >
      <h4>
        test02
      </h4>
    </div>
    <div
      className="card-answer"
    >
      <h4
        className="text-hidden"
      >
        answer02
      </h4>
    </div>
  </div>
</div>
`;
